File: remote/src/main/java/com/kelsos/mbrc/events/ui/TrackMoved.java
package com.kelsos.mbrc.events.ui;

import org.codehaus.jackson.node.ObjectNode;

public class TrackMoved {
  private boolean success;
  private int from;
  private int to;

  public TrackMoved(ObjectNode node) {
    success = node.path("success").asBoolean();
    from = node.path("from").asInt();
    to = node.path("to").asInt();
  }

  public boolean isSuccess() {
    return success;
  }

  public int getFrom() {
    return from;
  }

  public int getTo() {
    return to;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/events/ui/TrackMoved.kt
package com.kelsos.mbrc.events.ui

import com.fasterxml.jackson.databind.node.ObjectNode

class TrackMoved(node: ObjectNode) {
  val isSuccess: Boolean
  val from: Int
  val to: Int

  init {
    isSuccess = node.path("success").asBoolean()
    from = node.path("from").asInt()
    to = node.path("to").asInt()
  }
}
